modules/services/window-managers/i3-sway/lib/options.nix

```diff
@@ -831,4 +831,35 @@ in {
     '';
     example = "workspace number 9";
   };
+
+  workspaceOutputAssign = mkOption {
+    type = with types;
+      let
+        workspaceOutputOpts = submodule {
+          options = {
+            workspace = mkOption {
+              type = str;
+              default = "";
+              example = "Web";
+              description = ''
+                Name of the workspace to assign.
+              '';
+            };
+
+            output = mkOption {
+              type = str;
+              default = "";
+              example = "eDP";
+              description = ''
+                Name of the output from <command>
+                  ${if isSway then "swaymsg" else "i3-msg"} -t get_outputs
+                </command>.
+              '';
+            };
+          };
+        };
+      in listOf workspaceOutputOpts;
```

Comment:
What is the advantage of using `listOf` instead of `attrsOf str`?

Reply:
I'm unsure if I understand it right, but the assignments are a list of workspaceOutputOpts -> [ { workspace = "1"; output = "eDP"} ] so if I would change to attrsOf I would expect the need of a key? or would you change it to workspace as key and output as value?

Reply:
I think it would be natural to have this as workspace as the key, output as the value.

Reply:
In the end, listOf preserves the ordering so we should keep it that way.